app/controllers/groups_controller.rb

```diff
@@ -18,6 +18,31 @@ def show
     end
   end
 
+  def generate_token
+    @group = Group.where(id: params[:id])
+    @group.group_token_generate
```

Comment:
Above method will generate token for multiple groups
```suggestion
    @group = Group.find(params[:id])
    @group.regenerate_group_token
```

Reply:
what we are trying to achieve through this change?... this will change token everytime we click on button...

Reply:
In your current approach, it will reset token for multiple groups and is expensive, this change will limit it to just one group

Reply:
I tested it.. its running modal method is running only on the current group
here is the sql query
![2020-12-07_13-55](https://user-images.githubusercontent.com/55848322/101327095-362a7700-3894-11eb-9af6-9a8b5c6b1a6c.png)

Reply:
```
  @group = Group.find(params[:id])
    if (@group.updated_at <=12.days.ago or @group.group_token == nil)
      @group.regenerate_group_token
    end
```
instead of modal method i will use this...